lib/init/build.zig

```diff
@@ -113,4 +113,16 @@ pub fn build(b: *std.Build) void {
     const test_step = b.step("test", "Run unit tests");
     test_step.dependOn(&run_lib_unit_tests.step);
     test_step.dependOn(&run_exe_unit_tests.step);
+
+    // This creates a `docs` step. It will be visible in the `zig build --help`
+    // menu, and can be selected like this: `zig build docs`
+    // This will generate the package documentation from the doc comments.
+    const docs_step = b.step("docs", "Build the package documentation");
+    const docs_obj = b.addObject(.{ .name = ".NAME", .root_module = lib_mod });
+    const install_docs = b.addInstallDirectory(.{
+        .source_dir = docs_obj.getEmittedDocs(),
+        .install_dir = .prefix,
+        .install_subdir = "docs",
```

Comment:
This is a bit of an odd choice of directory. Since this is `zig init`, I think we should try to encourage best practice. I'd expect something more like `$prefix/share/doc/$name/html` (the norm on Debian/Ubuntu at least).

```suggestion
        .install_subdir = b.pathJoin(&.{ "share", "doc", ".NAME", "html" }),
```

Reply:
Autodocs output is hardly something distros would want to install so I don't think it makes sense to follow distro conventions here. Much more likely that someone will view the files from a local `zig-out/` or a CI system will copy them onto a static file server.

See also: https://github.com/ziglang/zig/blob/cb8d7a811003b08a7e16c533577fee92113585b1/build.zig#L60-L64

Reply:
> Autodocs output is hardly something distros would want to install

Hm, what makes you say that? I see an ocean of API docs in `/usr/share/doc` on my system, many of them in HTML form, and many of them generated from source code.

Reply:
Discoverability mostly - if it was a man page there is a tool that knows where to look for those. Windows back in the day had at least a chm viewer (which would be lovely for this now that I think about it!). If I have to manually dig around in some system directory where I may or may not find what I'm looking for I'll directly go to the nearest search engine instead.

Maybe I'm an outlier though 🙂

Reply:
I can't say I'm a frequent browser of `/usr/share/doc` myself... but my inclination is to follow established practice in distros when it comes to stuff like this.

@andrewrk any thoughts on this since you've done Debian packaging work?